core/Pos/Util/Util.hs

```diff
@@ -80,6 +81,13 @@ instance FromJSON Byte where
 instance ToJSON Byte where
     toJSON = toJSON . toBytes
 
+instance IsString s => MonadFail (Either s) where
```

Comment:
Probably write `TODO` here to not forgot remove this instance while in `base`: https://ghc.haskell.org/trac/ghc/ticket/12160